modules/base/src/properties/bufferinformationproperty.cpp

```diff
@@ -0,0 +1,202 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/base/properties/bufferinformationproperty.h>
+
+#include <inviwo/core/datastructures/buffer/buffer.h>
+#include <inviwo/core/util/stringconversion.h>
+
+namespace inviwo {
+
+const std::string BufferInformationProperty::classIdentifier =
+    "org.inviwo.BufferInformationProperty";
+std::string BufferInformationProperty::getClassIdentifier() const { return classIdentifier; }
+
+const std::string MeshBufferInformationProperty::classIdentifier =
+    "org.inviwo.MeshBufferInformationProperty";
+std::string MeshBufferInformationProperty::getClassIdentifier() const { return classIdentifier; }
+
+const std::string IndexBufferInformationProperty::classIdentifier =
+    "org.inviwo.IndexBufferInformationProperty";
+std::string IndexBufferInformationProperty::getClassIdentifier() const { return classIdentifier; }
+
+BufferInformationProperty::BufferInformationProperty(std::string identifier,
+                                                     std::string displayName,
+                                                     InvalidationLevel invalidationLevel,
+                                                     PropertySemantics semantics)
+    : CompositeProperty(identifier, displayName, invalidationLevel, semantics)
+    , format_("format", "Format", "")
+    , size_("size", "Size", 0, 0, std::numeric_limits<size_t>::max(), 1,
+            InvalidationLevel::InvalidOutput, PropertySemantics("Text"))
+    , usage_("usage", "Usage")
+    , target_("target", "Target") {
+
+    util::for_each_in_tuple(
```

Comment:
For each argument is a bit simpler

Reply:
sticking with `for_each_in_tuple` and `props()`:
```c++
        util::for_each_in_tuple([](auto& dst, auto& src) { dst = src; }, props(), that.props());
```